integration/spark2/src/main/scala/org/apache/spark/sql/secondaryindex/command/SICreationCommand.scala

```diff
@@ -0,0 +1,574 @@
+
+class ErrorMessage(message: String) extends Exception(message) {
+}
+
+ /**
+  * Command for index table creation
+  * @param indexModel      SecondaryIndex model holding the index infomation
+  * @param tableProperties SI table properties
+  * @param isCreateSIndex  if false then will not create index table schema in the carbonstore
+   *                        and will avoid dataload for SI creation.
+  */
+ private[sql] case class CreateIndexTable(indexModel: SecondaryIndex,
+     tableProperties: scala.collection.mutable.Map[String, String],
+     var isCreateSIndex: Boolean = true)
+   extends DataCommand {
+
+   val LOGGER: Logger = LogServiceFactory.getLogService(this.getClass.getCanonicalName)
+
+   override def processData(sparkSession: SparkSession): Seq[Row] = {
+    val databaseName = CarbonEnv.getDatabaseName(indexModel.databaseName)(sparkSession)
+    indexModel.databaseName = Some(databaseName)
+    val tableName = indexModel.tableName
+    val storePath = CarbonProperties.getStorePath
+    val dbLocation = CarbonEnv.getDatabaseLocation(databaseName, sparkSession)
+    val indexTableName = indexModel.indexTableName
+
+    val tablePath = dbLocation + CarbonCommonConstants.FILE_SEPARATOR + indexTableName
+     setAuditTable(databaseName, indexTableName)
+     setAuditInfo(Map("Column names" -> indexModel.columnNames.toString(),
+       "Parent TableName" -> indexModel.tableName,
+       "SI Table Properties" -> tableProperties.toString()))
+    LOGGER.info(
+      s"Creating Index with Database name [$databaseName] and Index name [$indexTableName]")
+    val catalog = CarbonEnv.getInstance(sparkSession).carbonMetaStore
+    val identifier = TableIdentifier(tableName, indexModel.databaseName)
+    var carbonTable: CarbonTable = null
+    var locks: List[ICarbonLock] = List()
+    var oldIndexInfo = ""
+
+    try {
+      carbonTable = CarbonEnv.getCarbonTable(indexModel.databaseName, tableName)(sparkSession)
+      if (carbonTable == null) {
+        throw new ErrorMessage(s"Parent Table $databaseName.$tableName is not found")
+      }
+
+      if (carbonTable != null &&
+          (carbonTable.isFileLevelFormat || !carbonTable.getTableInfo.isTransactionalTable)) {
+        throw new MalformedCarbonCommandException(
+          "Unsupported operation on non transactional table")
+      }
+
+      if (carbonTable.isStreamingSink) {
+        throw new ErrorMessage(
+          s"Parent Table  ${ carbonTable.getDatabaseName }." +
+          s"${ carbonTable.getTableName }" +
+          s" is Streaming Table and Secondary index on Streaming table is not supported ")
+      }
+
+      if (carbonTable.isHivePartitionTable) {
+        throw new ErrorMessage(
+          s"Parent Table  ${ carbonTable.getDatabaseName }." +
+          s"${ carbonTable.getTableName }" +
+          s" is Partition Table and Secondary index on Partition table is not supported ")
```

Comment:
is it easy to support it in the future?

Reply:
yes, there will some issues in pruning part, but we can support